values/src/dimension.rs

```diff
@@ -490,44 +491,170 @@ pub enum Ambiguity {
     Big,
 }
 
-/// Payload for the time of Dimension
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Copy)]
+pub enum DatetimeKind {
+    Date,
+    Time,
+    DatePeriod,
+    TimePeriod,
+    DatetimeComplement,
+    Datetime,
+    Empty,
+}
+
+
+/// Payload for the datetime value of Dimension
 #[derive(Clone)]
-pub struct TimeValue {
+pub struct DatetimeValue {
     pub constraint: RcConstraint<Local>,
     pub form: Form,
     pub direction: Option<BoundedDirection>,
     pub precision: Precision,
     pub latent: bool,
     pub ambiguity: Ambiguity,
+    pub datetime_kind: DatetimeKind,
 }
 
 // We need partial eq to make Dimension partial eq happy, but this is only
 // useful for testing.
-impl PartialEq for TimeValue {
-    fn eq(&self, _other: &TimeValue) -> bool {
+impl PartialEq for DatetimeValue {
+    fn eq(&self, _other: &DatetimeValue) -> bool {
         unimplemented!()
     }
 }
 
-impl ::std::fmt::Debug for TimeValue {
+impl ::std::fmt::Debug for DatetimeValue {
     fn fmt(&self, fmt: &mut ::std::fmt::Formatter) -> ::std::result::Result<(), ::std::fmt::Error> {
-        write!(fmt, "<TimeValue>")
+        write!(fmt, "DatetimeValue(form={:?}, grain={:?}, min-grain={:?}, kind={:?})",
+               self.form, self.constraint.grain(), self.constraint.grain_min(), self.datetime_kind)
+
     }
 }
 
-impl TimeValue {
+impl DatetimeValue {
     pub fn is_coarse_grain_smaller_than(&self, grain: Grain) -> bool {
         (self.constraint.coarse_grain_step() as usize) < (grain as usize)
     }
 
     pub fn is_coarse_grain_greater_than(&self, grain: Grain) -> bool {
         (self.constraint.coarse_grain_step() as usize) > (grain as usize)
     }
+
+    pub fn date_form(&self) -> Option<bool> {
+        match self.form {
+            Form::Cycle(grain) => {
+                match grain {
+                    Grain::Year => Some(true),
+                    Grain::Quarter => Some(true),
+                    Grain::Month => Some(true),
+                    Grain::Week => Some(true),
+                    Grain::Day => Some(true),
+                    _ => Some(false),
+                }
+            },
+            Form::Year(_) => Some(true),
+            Form::Month(_) => Some(true),
+            Form::MonthDay(_) => Some(true),
+            Form::YearMonthDay(_) => Some(true),
+            Form::TimeOfDay(_) => Some(false),
+            Form::DayOfWeek { .. } => Some(true),
+            Form::Empty => None,
+            Form::PartOfDay { .. } => Some(false),
+            Form::Meal => Some(false),
+            Form::Celebration => Some(true),
+            Form::PartOfMonth => Some(true),
+            Form::PartOfYear => Some(true),
+            Form::DayOfMonth => Some(true),
+            Form::PartOfForm(_) => None,
+            Form::PartOfWeek => Some(true),
+            Form::Span => None,
+        }
+    }
+
+    pub fn time_form(&self) -> Option<bool> {
```

Comment:
`is_time_form` would be better as this returns a boolean value.

Reply:
✅